conformance/tests/httproute-listener-hostname-matching.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: httproute-listener-hostname-matching
+  namespace: gateway-conformance-infra
+spec:
+  gatewayClassName: {GATEWAY_CLASS_NAME}
+  listeners:
+  - name: listener-1
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: Same
+    hostname: bar.com
+  - name: listener-2
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: Same
+    hostname: foo.bar.com
+  - name: listener-3
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: Same
+    hostname: "*.bar.com"
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: backend-v1
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: httproute-listener-hostname-matching
+    namespace: gateway-conformance-infra
+    sectionName: listener-1
+  rules:
+  - backendRefs:
+    - name: infra-backend-v1
+      port: 8080
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: backend-v2
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: httproute-listener-hostname-matching
+    namespace: gateway-conformance-infra
+    sectionName: listener-2
+  rules:
+  - backendRefs:
+    - name: infra-backend-v2
+      port: 8080
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
```

Comment:
Should we add some HTTPRoutes with hostname as well to test intersection?

Especially something like listener: `*.com`, route: `foo.example.com`

Reply:
I have another test coming that specifically tests hostname intersections -- I thought it was simpler to split them into separate tests, but can combine them if folks prefer.

Reply:
👍 Agree that it would be good to have. I'd also tend to lean towards a separate PR for the additional tests, but don't have a strong preference.